scripts/env-doctor.sh

```diff
@@ -0,0 +1,121 @@
+#!/bin/bash
+
+set -e
+
+# Colors for output
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+YELLOW='\033[0;33m'
+BLUE='\033[0;34m'
+NC='\033[0m' # No Color
+
+# Function to run a diagnostic and its remedy
+run_diagnostic() {
+    local description="$1"
+    local diagnostic="$2"
+    local remedy_description="$3"
+    local remedy="$4"
+
+    echo -e "${YELLOW}Diagnostic:${NC} $description"
+    if eval "$diagnostic"; then
+        echo -e "${GREEN}✓ Passed${NC}"
+    else
+        echo -e "${RED}✗ Failed${NC}"
+        echo -e "${YELLOW}Remedy:${NC} $remedy_description"
+        if [[ $AUTO_YES != "true" ]]; then
+            read -p "Do you want to apply this remedy? (Yes/No/All) " answer
+            case ${answer:0:1} in
+                y|Y) ;;
+                a|A) AUTO_YES=true ;;
+                *) return ;;
+            esac
+        fi
+        if ! eval "$remedy"; then
+            echo -e "${RED}Remedy failed. Please seek more help.${NC}"
+            exit 1
+        fi
+        echo "Remedy applied."
+    fi
+    echo
+}
+
+# Diagnostics and remedies
+run_diagnostic \
+    "Check if fnm is installed" \
+    "command -v fnm >/dev/null 2>&1" \
+    "Install fnm" \
+    "brew install fnm"
+
+run_diagnostic \
+    "Check if Node.js is installed" \
+    "command -v node >/dev/null 2>&1" \
+    "Install Node.js using fnm" \
+    "fnm install --lts && fnm use lts-latest"
+
+run_diagnostic \
+    "Check if Node.js version matches .node-version" \
+    "[ \"$(node --version)\" = \"v$(cat .node-version)\" ]" \
+    "Install the correct Node.js version using fnm" \
+    "fnm install $(cat .node-version) && fnm use $(cat .node-version)"
+
+run_diagnostic \
+    "Check if Yarn version matches package.json" \
+    "[ \"$(yarn --version)\" = \"$(node -p "require('./package.json').packageManager.split('@')[1]")\" ]" \
+    "Install correct Yarn version" \
+    "corepack enable && yarn set version $(node -p "require('./package.json').packageManager.split('@')[1]")"
+
+run_diagnostic \
+    "Check if Git is installed" \
+    "command -v git >/dev/null 2>&1" \
+    "Install Git" \
+    "brew install git"
+
+run_diagnostic \
+    "Check if VSCode is installed" \
+    "command -v code >/dev/null 2>&1" \
+    "Install Visual Studio Code" \
+    "brew install --cask visual-studio-code"
+
+run_diagnostic \
+    "Check if Docker is installed" \
+    "command -v docker >/dev/null 2>&1" \
+    "Install Docker" \
+    "brew install --cask docker"
+
+run_diagnostic \
+    "Build repo" \
+    "yarn build" \
+    "Clean, reinstall dependencies, and rebuild" \
+    "git clean -fdx && yarn install && yarn build"
```

Comment:
My main concern with this one is that it nukes any `.vscode` or other IDE config files. That might not work for everyone, at least not without warning.

Reply:
True. It does have a warning but probably not severe enough. It could also specifically blow up `node_modules` rather than everything gitignored. I'll try changing this to to separate ones: reset node_modules (for `yarn install`) and reset bundles (for `yarn build`).